v3/src/components/tool-shelf/tiles-list-button.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react"
+import { observer } from "mobx-react-lite"
+import { Menu, MenuButton, MenuItem, MenuList } from "@chakra-ui/react"
+import OptionsIcon from "../../assets/icons/icon-options.svg"
+import { useDocumentContent } from "../../hooks/use-document-content"
+import { uiState } from "../../models/ui-state"
+import { IFreeTileLayout } from "../../models/document/free-tile-row"
+import { t } from "../../utilities/translation/translate"
+import { kRightButtonBackground, ToolShelfButtonTag } from "./tool-shelf-button"
+import GraphIcon from "../../assets/icons/icon-graph.svg"
+import TableIcon from "../../assets/icons/icon-table.svg"
+import SliderIcon from "../../assets/icons/icon-slider.svg"
+import MapIcon from "../../assets/icons/icon-map.svg"
+import CalculatorIcon from "../../assets/icons/icon-calc.svg"
+import GuideIcon from "../../assets/icons/icon-guide.svg"
+import WebViewIcon from "../../assets/icons/icon-media-tool.svg"
+
+import "./tool-shelf.scss"
+
+export const TilesListShelfButton = observer(function TilesListShelfButton() {
+  const documentContent = useDocumentContent()
+  const tilesArr: any[] = []
+
+  documentContent?.tileMap.forEach((tile) => {
+    tilesArr.push(tile)
+  })
+
+  const tileTypeIconMap = [{type: "Graph", icon: <GraphIcon className="tile-list-icon"/>},
+                            {type: "CaseTable", icon: <TableIcon className="tile-list-icon"/>},
+                            {type: "CodapSlider", icon: <SliderIcon className="tile-list-icon"/>},
+                            {type: "Map", icon: <MapIcon className="tile-list-icon"/>},
+                            {type: "Calculator", icon: <CalculatorIcon className="tile-list-icon"/>},
+                            {type: "Guide", icon: <GuideIcon className="tile-list-icon"/>},
+                            {type: "CodapWebView", icon: <WebViewIcon className="tile-list-icon"/>}
+                          ]
```

Comment:
This should just be a map, like
```
const tileTypeIconMap = {
  Graph: <GraphIcon className="tile-list-icon" />,
  ...
}
```

Better yet, you could add these icons to the `gTileComponentInfoMap` via `registerTileComponentInfo`. Looking at this, it seems there's already an `Icon` specified here... are these icons different than the ones you're using? If they're the same, you should remove all of this and just use `getTileComponentIcon` instead. But even if they're different, I think it would be better to use that registration system for this.